src/as/core/BufferGeometry.ts

```diff
@@ -71,7 +80,7 @@ export class BufferGeometry extends EventDispatcher {
 	morphAttributes: Map<string, BufferAttribute[]> = new Map()
 	//^ per BufferGeometryLoader.js in the original three.js, geometry.morphAttributes[key] is loaded with array of BufferAttributes
 
-	groups: BufferGeometryGroup[] = []
+	groups: BufferGeometryGroup[] = new Array<BufferGeometryGroup>()
```

Comment:
`groups: BufferGeometryGroup[] = []` is the same thing, but less characters.

When TS (and AS) sees `[]` being assigned to an LHS that has the type declaration, then the RHS automatically has that type. It is type safe. It's just syntax short hand.

Reply:
In plain JS/TS both are the exact same (at runtime, compile time, and type check time). But dcode [answered here](https://discord.com/channels/721472913886281818/721497900932137090/787930839861821460) that using array literals is nice because it will try to optimize into a `StaticArray` under the hood (still appears as an `Array` to the code writer), and fall back to regular `Array` for dynamic length if something like `array.push` is used.

So basically if we use the literal format, we may sometimes get an optimization. This is good to know!

This means, for things like the `elements` array inside of `Matrix4` that contains 16 numbers, the length will never change, so initializing it as `[n1, n2, ..., n16]` will automatically make it a `StaticArray` without all the dynamic stuff that `Array` has. 

Neat!

Reply:
In our case here, we are making empty `[]` arrays, so they will end up as `Array` because we need to expand their size later.

But TLDR, we can use `[]` for syntax short-hand, without penalty. If your IDE was showing any type error relating to `any[]`, we should see what is up with that. It works well on my end.

Reply:
StaticArray is only used when you specify it because it has a different memory layout than Array. Using `[]` instead of `new Array()` is nicer though.

Reply:
good to know thanx